### PR TITLE
Implements QueryResults.__eq__

### PR DESCRIPTION
**Pull request by funkyfuture**

### CODE REVIEW

File: _delb/nodes.py [hunk 1]
```diff
@@ -2782,6 +2782,12 @@ class QueryResults(Sequence[TagNode]):
     def __init__(self, results: Iterator[_ElementWrappingNode]):
         self.__items = cast(Tuple[TagNode], tuple(results))
 
+    def __eq__(self, other):
+        if not isinstance(other, QueryResults):
```
Comment (funkyfuture) [inline _delb/nodes.py:2786]: not sure whether all Sequences should be supported.

File: _delb/nodes.py [hunk 2]
```diff
@@ -2782,6 +2782,12 @@ class QueryResults(Sequence[TagNode]):
     def __init__(self, results: Iterator[_ElementWrappingNode]):
         self.__items = cast(Tuple[TagNode], tuple(results))
```
Comment (funkyfuture) [inline _delb/nodes.py:2783]: very btw, shouldn't this be `Tuple[TagNode, ...]`?

File: _delb/nodes.py (hunk 1 above)
Comment (JKatzwinkel) [reply _delb/nodes.py:2786]: why not?

Comment (funkyfuture) [reply _delb/nodes.py:2786]: it'd require more tests… i'm not in the mood.

Comment (funkyfuture) [reply _delb/nodes.py:2786]: and i wonder what significance order has.

File: tests/test_queries.py [hunk 3]
```diff
@@ -178,6 +178,35 @@ def test_results_as_other_type(queries_sample):
...
+                <w corresp="src:tlaIBUBdQbsUzWoU0ZAg6KyIT74EPU"><unclear/></w>
+                <w corresp="src:tlaIBUBd4C0OAENyE3Ti62GjqGFmto"><unclear/></w>
+                <w corresp="src:tlaIBUBd8XAOdLv8k7WiQ1f7ZFe3IQ"><unclear/></w>
+                <w corresp="src:tlaIBUBd37vHGdUYkiYqNF8ExKiO6M"><unclear/></w>
+            </s>
+        </root>
+        """
+    )
+    words = document.css_select("s w")
+    assert words == words.filtered_by(lambda n: True)
```
Comment (JKatzwinkel) [inline tests/test_queries.py:207]:
```suggestion
    assert words == document.css_select("s w")
    assert words == words.filtered_by(lambda n: True)
```

File: tests/test_queries.py [hunk 4]
```diff
@@ -178,6 +178,35 @@ def test_results_as_other_type(queries_sample):
...
+                <w corresp="src:tlaIBUBd4NQUh0FikJ0stCGrcxq9wk"><unclear/></w>
+                <w corresp="src:tlaIBUBd7VHAfkj20bDsv3QzaQ4eoo"><unclear/></w>
+                <w corresp="src:tlaIBUBd5L0JpJVZUOWoGFGhqXAfqc"><unclear/></w>
+                <w corresp="src:tlaIBUBd9zrhmbxrkyqpG7t84kiw2s"><unclear/></w>
+                <w corresp="src:tlaIBUBd18UraAwdkCUgzqNsZplqIw"><unclear/></w>
+                <w corresp="src:tlaIBUBd8EjhpjvSERfoCg6pz5qYxc"><unclear/></w>
+                <w corresp="src:tlaIBUBdQbsUzWoU0ZAg6KyIT74EPU"><unclear/></w>
+                <w corresp="src:tlaIBUBd4C0OAENyE3Ti62GjqGFmto"><unclear/></w>
+                <w corresp="src:tlaIBUBd8XAOdLv8k7WiQ1f7ZFe3IQ"><unclear/></w>
+                <w corresp="src:tlaIBUBd37vHGdUYkiYqNF8ExKiO6M"><unclear/></w>
```
Comment (JKatzwinkel) [inline tests/test_queries.py:201]:
```suggestion
                <w corresp="src:tlaIBUBdzQ3wWIW60TVhNy3cRxYmgg"><unclear/></w>
                <w corresp="src:tlaIBUBd7n0fy1OPU1DjVU66j2B4Qc"><unclear/></w>
```

File: _delb/nodes.py (hunk 2 above)
Comment (JKatzwinkel) [reply _delb/nodes.py:2783]: I guesz...

File: _delb/nodes.py (hunk 1 above)
Comment (JKatzwinkel) [reply _delb/nodes.py:2786]: ok good point

**Review by JKatzwinkel — DISMISSED** [review]

**Comment by JKatzwinkel** [comment]
fixes #37

**Comment by funkyfuture** [comment]
i think this isn't worth to be lsted in the changlog.

would a soonish release help you?

File: _delb/nodes.py (hunk 1 above)
Comment (funkyfuture) [reply _delb/nodes.py:2786]: i unreoslved this b/c i wanna give it some thought.

Comment (funkyfuture) [reply _delb/nodes.py:2786]: luckily [the words](https://www.w3.org/TR/1999/REC-xpath-19991116/#section-Introduction) are very clear right at the beginning:

> node-set (an unordered collection of nodes without duplicates)

i haven't read until the end whether they came up with the idea to coerce that into another type.

hence it should be compared like a set. but what if the other is a list that contains a node twice?

**Comment by funkyfuture** [comment]
it's good enough for me now.

File: tests/test_queries.py [hunk 5]
```diff
@@ -178,6 +178,27 @@ def test_results_as_other_type(queries_sample):
...
+        """
+    )
+    word_nodes = document.css_select("s w")
+    assert word_nodes == word_nodes.filtered_by(lambda n: True)
+    assert word_nodes == word_nodes.as_list()
+    assert word_nodes == tuple(reversed(word_nodes.as_list()))
+    assert word_nodes != 2 * word_nodes.as_list()
+    assert [document.root] == document.css_select('root')
+    with pytest.raises(TypeError):
+        document.css_select('root') == document.root
```
Comment (funkyfuture) [inline tests/test_queries.py:199]:
```suggestion
        document.css_select("root") == document.root
```

File: tests/test_queries.py [hunk 6]
```diff
@@ -178,6 +178,27 @@ def test_results_as_other_type(queries_sample):
...
+            </s>
+        </root>
+        """
+    )
+    word_nodes = document.css_select("s w")
+    assert word_nodes == word_nodes.filtered_by(lambda n: True)
+    assert word_nodes == word_nodes.as_list()
+    assert word_nodes == tuple(reversed(word_nodes.as_list()))
+    assert word_nodes != 2 * word_nodes.as_list()
+    assert [document.root] == document.css_select('root')
```
Comment (funkyfuture) [inline tests/test_queries.py:197]:
```suggestion
    assert [document.root] == document.css_select("root")
```

File: _delb/nodes.py [hunk 7]
```diff
@@ -2780,7 +2780,13 @@ class QueryResults(Sequence[TagNode]):
...
 
     def __init__(self, results: Iterator[_ElementWrappingNode]):
-        self.__items = cast(Tuple[TagNode], tuple(results))
+        self.__items = cast(Tuple[TagNode, ...], tuple(results))
+
+    def __eq__(self, other):
+        if not isinstance(other, Collection):
+            raise TypeError
+
+        return len(self.__items) == len(other) and all(x in other for x in self.__items)
```
Comment (JKatzwinkel) [inline _delb/nodes.py:2789]: tbh I don't understand why this works.. how does python know to correctly determine `x in other` if `other` is of type `QueryResults`, meaning it has to check whether `x in other.__items` even though there is no `QueryResults.__contains__` that tells it that!!?

**Review by JKatzwinkel — DISMISSED** [review]

File: _delb/nodes.py (hunk 7 above)
Comment (JKatzwinkel) [reply _delb/nodes.py:2789]: ah [ok](https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes).

**Comment by funkyfuture** [comment]
are you on station where you can rebase onto the `main` branch and force-push the result (squashed or not)? i'd merge it then w/o approval.

**Review by JKatzwinkel — APPROVED** [review]

**Comment by funkyfuture** [comment]
cool.

**Pull request merged**